engine/apps/alerts/models/alert.py

```diff
@@ -79,28 +81,34 @@ def get_integration_optimization_hash(self):
     @classmethod
     def create(
         cls,
-        title,
-        message,
-        image_url,
-        link_to_upstream_details,
-        alert_receive_channel,
-        integration_unique_data,
-        raw_request_data,
+        title: typing.Optional[str],
+        message: typing.Optional[str],
+        image_url: typing.Optional[str],
+        link_to_upstream_details: typing.Optional[str],
+        alert_receive_channel: "AlertReceiveChannel",
+        integration_unique_data: typing.Optional[typing.Dict],
+        raw_request_data: RawRequestData,
         enable_autoresolve=True,
-        is_demo=False,
-        channel_filter=None,
-        force_route_id=None,
-        received_at=None,
-    ):
+        is_demo: bool = False,
+        channel_filter: typing.Optional["ChannelFilter"] = None,
+        force_route_id: typing.Optional[int] = None,
+        received_at: typing.Optional[str] = None,
+    ) -> "Alert":
         """
         Creates an alert and a group if needed.
         """
         # This import is here to avoid circular imports
         from apps.alerts.models import AlertGroup, AlertGroupLogRecord, AlertReceiveChannel, ChannelFilter
 
+        parsed_labels = gather_labels_from_alert_receive_channel_and_raw_request_data(
+            alert_receive_channel, raw_request_data
+        )
```

Comment:
render_group_data() on line 106 is parsing group_id, etc out of alert and stores in AlertGroup.GroupData, so it makes sense to parse labels there as well

Reply:
good call, done 👍